pkg/rtc/datatrack.go

```diff
@@ -0,0 +1,186 @@
+package rtc
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/livekit/livekit-server/pkg/sfu"
+	"github.com/livekit/protocol/livekit"
+	"github.com/livekit/protocol/logger"
+	"github.com/pion/webrtc/v3"
+	"google.golang.org/protobuf/proto"
+)
+
+type DataTrackSender interface {
+	sfu.TrackSender
+	Write(label string, data []byte)
+}
+
+type DataTrack struct {
+	trackID       livekit.TrackID
+	participantID livekit.ParticipantID
+	logger        logger.Logger
+	lock          sync.RWMutex
+	downTracks    []DataTrackSender
+	onDataPacket  func(*livekit.DataPacket)
+	onClose       []func()
+}
+
+func NewDataTrack(trackID livekit.TrackID, participantID livekit.ParticipantID, logger logger.Logger) *DataTrack {
+	t := &DataTrack{
+		trackID:       trackID,
+		participantID: participantID,
+		logger:        logger,
+	}
+	return t
+}
+
+func (t *DataTrack) onData(label string, data []byte) {
+	t.lock.RLock()
+	f := t.onDataPacket
+	dts := t.downTracks
+	t.lock.RUnlock()
+	for _, dt := range dts {
+		dt.Write(label, data)
+	}
+	if f != nil {
```

Comment:
can we add an empty line here please? Would be good to have visual separate between parts.